### PR TITLE
Implement syntactic support for GNU's `__builtin_choose_expr`

### PR DESCRIPTION
**Pull request by ludanpr**
Issue #69

### CODE REVIEW

File: C/syntax/SyntaxKind.h [hunk 1]
```diff
@@ -409,6 +410,7 @@ enum SyntaxKind : std::uint16_t
     CompoundLiteralExpression,
     CastExpression,
     VAArgumentExpression,
+    ExtGNU_ChooseExpr,
```
Comment (ltcmelo) [inline C/syntax/SyntaxKind.h:413]: Should be `ExtGNU_ChooseExpression` for reason similar to that of the first comment.

File: C/syntax/SyntaxNodes_Expressions.h [hunk 2]
```diff
@@ -743,6 +743,49 @@ class PSY_C_API VAArgumentExpressionSyntax final : public ExpressionSyntax
...
 
+/**
+ * \brief The ExtGNU_ChooseExprSyntax class.
+ *
+ * \code
+ * __builtin_choose_expr( const_expr, expr1, expr2 )
+ * \endcode
+ *
+ * \remark \c const_expr is an integer constant expression.
+ */
```
Comment (ltcmelo) [inline C/syntax/SyntaxNodes_Expressions.h:754]: Could you add a note for the corresponding AST in clang/llvm (like the notes that exist in other classes)?

```
* \note Similar to:
* - \c clang::ChooseExpr of LLVM/Clang.
 ```

File: C/syntax/SyntaxVisitor.h [hunk 3]
```diff
@@ -144,6 +144,7 @@ class PSY_C_API SyntaxVisitor
     virtual Action visitConditionalExpression(const ConditionalExpressionSyntax*) { return Action::Visit; }
     virtual Action visitAssignmentExpression(const AssignmentExpressionSyntax*) { return Action::Visit; }
     virtual Action visitSequencingExpression(const SequencingExpressionSyntax*) { return Action::Visit; }
+    virtual Action visitExtGNU_ChooseExpr(const ExtGNU_ChooseExprSyntax*) { return Action::Visit; }
```
Comment (ltcmelo) [inline C/syntax/SyntaxVisitor.h:147]: Should be `visitExtGNU_ChooseExpression`.

File: C/tests/TestParser_1000_1999.cpp [hunk 4]
```diff
@@ -2633,10 +2633,74 @@ void TestParser::case1967() {}
 void TestParser::case1968() {}
 void TestParser::case1969() {}
 
-void TestParser::case1970() {}
-void TestParser::case1971() {}
-void TestParser::case1972() {}
-void TestParser::case1973() {}
+void TestParser::case1970()
```
Comment (ltcmelo) [inline C/tests/TestParser_1000_1999.cpp:2636]: For the parser tests I like to separate each token by a space, e.g. `__builtin_choose_expr ( 0 , x , y )` and `__builtin_choose_expr ( 1 , sizeof ( int ) , sizeof ( double ) )`.

This helps  me focusing on syntax purely, without associating semantics to it — given that ambiguity is important for psyche-c. It also makes grep consistent "everywhere" (which also is the reason why I follow the convention of `x` for the first identifier, `y` for the second, etc).

**Review by ltcmelo — CHANGES_REQUESTED** [review]
Thanks @ludanpr , the PR is good! I could even merge it as is, but, if possible, it'd be great if you could adjust the minor comments that I left. Is that OK?

File: C/Fwds.h [hunk 5]
```diff
@@ -143,6 +143,7 @@ class BinaryExpressionSyntax;
 class ConditionalExpressionSyntax;
 class AssignmentExpressionSyntax;
 class SequencingExpressionSyntax;
+class ExtGNU_ChooseExprSyntax;
```
[the remark this answers is not in the document]
Comment (ludanpr) [reply C/Fwds.h:146]: :heavy_check_mark:

File: C/syntax/SyntaxNodes_Expressions.h (hunk 2 above)
Comment (ludanpr) [reply C/syntax/SyntaxNodes_Expressions.h:754]: :heavy_check_mark:

**Comment by ltcmelo** [comment]
Thank you @ludanpr ! You'll get another bounty :-)

**Pull request merged**